agent/command_test.go

```diff
@@ -328,20 +330,61 @@ functions:
     params:
       updates: 
       - key: key1
-        value: "expansion-update-value"
+        value: ${key1}
+      - key: key2
+        value: ${key2}
 `
 	s.setUpConfigAndProject(projYml)
 
 	func1 := model.PluginCommandConf{
 		Function:    "yes",
 		DisplayName: "function",
-		Vars:        map[string]string{"key1": "functionVar"},
+		Vars:        map[string]string{"key1": "functionVar1", "key2": "functionVar2", "key3": "functionVar3"},
+	}
+
+	cmds := []model.PluginCommandConf{func1}
+	err := s.a.runCommandsInBlock(s.ctx, s.tc, cmds, runCommandsOptions{}, "")
+	s.NoError(err)
+
+	key1Value := s.tc.taskConfig.Expansions.Get("key1")
+	s.Equal("functionVar1", key1Value, "key1 should be set to what it was updated to with expansions.update")
+
+	key2Value := s.tc.taskConfig.Expansions.Get("key2")
+	s.Equal("functionVar2", key2Value, "key2 should be set to what it was updated to with expansions.update")
+
+	key3Value := s.tc.taskConfig.Expansions.Get("key3")
+	s.Equal("expansionVar3", key3Value, "key3 should be the original expansion value")
+
+	s.Equal(&util.Expansions{}, s.tc.taskConfig.DynamicExpansions)
+}
+
+func (s *CommandSuite) TestVarsUnsetPreserveExpansionUpdatesFromFile() {
+	projYml := `
+functions:
+  yes:
+    command: expansions.update
+    params:
+      file: command/testdata/git/test_expansions.yml
+`
+	s.setUpConfigAndProject(projYml)
+
+	func1 := model.PluginCommandConf{
+		Function:    "yes",
+		DisplayName: "function",
+		Vars:        map[string]string{"key1": "newValue1", "key2": "newValue2", "key3": "newValue3"},
 	}
 
 	cmds := []model.PluginCommandConf{func1}
 	err := s.a.runCommandsInBlock(s.ctx, s.tc, cmds, runCommandsOptions{}, "")
 	s.NoError(err)
 
 	key1Value := s.tc.taskConfig.Expansions.Get("key1")
-	s.Equal("expansion-update-value", key1Value, "key1 should be set to what it was updated to with expansions.update")
+	s.Equal("newValue1", key1Value, "key1 should be set to what it was updated to with expansions.update")
+
+	key2Value := s.tc.taskConfig.Expansions.Get("key2")
+	s.Equal("newValue2", key2Value, "key2 should be set to what it was updated to with expansions.update")
+
+	key3Value := s.tc.taskConfig.Expansions.Get("key3")
+	s.Equal("expansionVar3", key3Value, "key3 should be the original expansion value")
+	s.Equal(&util.Expansions{}, s.tc.taskConfig.DynamicExpansions)
```

Comment:
nit: you can just check `s.Empty` rather than comparing it to the empty struct (same with the others).